### ISSUE TITLE
Update juggernaut to use latest version of socket.io library

### ISSUE THREAD

**Issue by ivanpoznyak** [maccman/juggernaut#81]
Hi there.

Socket io has recently been updated with quite important fix for JSONPPolling - see https://github.com/LearnBoost/Socket.IO/commit/a7e1a34d6b5bceb7030eecf135ec6569ab300aa8.

The issue was that in case of reconnection - Juggernaut was unusable in IE. This happened because it still used old session id when resubscribing to different channels etc. thus making server unable to push updates to IE clients, though the client itself considered himself connected.

Can you upgrade to new version of the lib or provide a timeframe when this will happen?

**Comment by maccman** [comment]
Yes, we're planning on updating within the next 2 weeks. Pull requests are always welcome.

**Comment by inspire22** [comment]
any way to disable jsonp-polling in the meanwhile?

**Comment by maccman** [comment]
Yes, I believe you can pass a list of transports you want to use to
Socket.IO.

On Thu, Jun 23, 2011 at 11:50 AM, inspire22 <
reply@reply.github.com>wrote:

> any way to disable jsonp-polling in the meanwhile?
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/maccman/juggernaut/issues/81#issuecomment-1427061

## 

Alex MacCaw

+12147175129
@maccman

http://alexmaccaw.co.uk | http://www.leadthinking.com | http://socialmod.com

**Comment by fwoeck** [comment]
+1 an upgrade would be great!

**Comment by jer0m** [comment]
+1 for this update too!

Thanks!

**Comment by duncanbeevers** [comment]
Does it make sense to include Socket.IO as a submodule?
